lib/workers/repository/process/write.ts

```diff
@@ -31,6 +32,61 @@ export function canSkipBranchUpdateCheck(
   return true;
 }
 
+export function syncBranchState(
+  branchName: string,
+  baseBranch: string,
+  repositoryCache?: string
+): BranchCache {
+  const branchExisted = branchExists(branchName);
+  const branchSha = getBranchCommit(branchName)!;
+  const baseBranchSha = getBranchCommit(baseBranch)!;
+
+  const cache = getCache();
+  cache.branches ??= [];
+  const { branches: cachedBranches } = cache;
+  let branchState = cachedBranches.find((br) => br.branchName === branchName);
+  if (!branchState) {
+    if (branchExisted && repositoryCache === 'enabled') {
+      logger.debug(`No branch cache found for ${branchName}`);
+    }
+
+    // create a minimal branch state
+    branchState = {
+      branchName,
+      sha: branchSha,
+      baseBranch,
+      baseBranchSha,
+    } as BranchCache;
+    cachedBranches.push(branchState);
+  }
+
+  // if base branch name has changed invalidate cached isModified state
+  if (baseBranch !== branchState.baseBranch) {
+    branchState.baseBranch = baseBranch!;
+    delete branchState.isModified;
+  }
+
+  // if base branch sha has changed invalidate cache isBehindBase state
+  if (baseBranchSha !== branchState.baseBranchSha) {
+    delete branchState.isBehindBase;
```

Comment:
What about isConflicted?

Reply:
Haven't introduced it yet, we still use gitConflicts in this branch. Task for another PR.